### PR TITLE
chapter 10 txr

### PR DESCRIPTION
**Pull request by txr12**

### CODE REVIEW

File: chapter10/txr/ex1.pl [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/perl -w
+use 5.14.0;
+
+my $secret_number = get_random();
+my $exit;
+
+say 'enter a integer between 1 to 100 (included) ';
+say 'enter \'exit\' or \'quit\' or simply press enter to exit ';
+until ($exit) {
```
Comment (a-panayotopoulos) [inline chapter10/txr/ex1.pl]: `$exit` will never get a value, so you might as well use `while (1)`

File: chapter10/txr/ex1.pl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  chomp ( my $answer = <STDIN>);
+  last if is_exit_answer($answer); 
+  redo unless validate_number($answer);
```
Comment (a-panayotopoulos) [inline chapter10/txr/ex1.pl]: Nice touch.

File: chapter10/txr/ex2.pl [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/perl -w
+use 5.14.0;
+
+my $debug = $ARGV[0] // 1;
```
Comment (a-panayotopoulos) [inline chapter10/txr/ex2.pl]: Hah, I didn't realise you could do that without throwing an error. Nice.

File: chapter10/txr/ex3.pl [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+$ENV{ABC} = undef;
+my $max_length = max_length(keys(%ENV));
+
+foreach my $key ( sort (keys (%ENV))) {
+				printf "%-${max_length}s = %s\n",  $key, $ENV{$key}// '(undefined value)' ;
+}
+
+sub max_length {
+				return length(reduce { length ($a) > length ($b) ? $a : $b } @_);
+}
```
Comment (a-panayotopoulos) [inline chapter10/txr/ex3.pl]: All good.

File: chapter10/txr/ex1.pl (hunk 1 above)
Comment (txr12) [inline chapter10/txr/ex1.pl]: yup

**Pull request merged**